docs/pages/modules/shared-objects.mdx

```diff
@@ -0,0 +1,291 @@
+
+In this guide, let's understand what a shared object is and how they are implemented in native platforms.
+
+## What is a shared object?
+
+A shared object is a custom class that bridges a native instance from Android and/or iOS to your app's JavaScript/TypeScript code through an Expo module. On the native side in Kotlin and Swift, you declare the class by inheriting from `SharedObject` and expose it in your module definition using `Class()`. The Expo Modules API handles reference counting, lifecycle management, and cleanup automatically.
+
+## Why use shared objects?
+
+Large media assets, like images, can exceed several megabytes once decoded into memory. Without shared objects, passing these assets between different parts of your app forces each part to reload from the disk every time and decode the same file multiple times. This can increase memory pressure, create an I/O bottleneck, drop frames, or cause battery drain.
+
+Shared objects solve this by keeping a single native instance alive in memory while multiple JavaScript references point to it.
+
+## Example: Image manipulation without disk I/O
+
+To understand shared objects, consider an example where you need to rotate an image picked by the app user and then display that image in your app after manipulating it.
+
+### Without shared objects
+
+Historically, native modules were often written in a _stateless_ way, where each function operated independently without maintaining state between calls. If you wanted to perform two separate operations on the same object (such as an image file), you would load it from disk in both places and repeat the I/O operation each time.
+
+Without shared objects, `ImagePicker` returns a file URI such as `"file:///path/to/image.jpg"`. The image manipulator module then reads the URI and decodes the image from the disk into memory. When the app user calls `rotate()` method to rotate the image, the module saves the rotated image to a new file. Finally, when the new URI is passed to the `Image` component, it decodes the image from disk again to render the image. This workflow results in two or more decode and disk read operations.
+
+### With shared objects
+
+The same scenario becomes much more efficient with shared objects. `ImagePicker` returns a file URI, but the image manipulator module reads the URI and decodes the image once into a shared object. When the app user calls `rotate()`, the module manipulates the in-memory bitmap without writing to the disk.
+
+Finally, the new URI is passed to the `Image` component and this time the image is rendered from memory. The entire workflow requires just one disk read and one decode operation, with all transformations happening in memory.
+
+The performance gains are significant. By eliminating redundant disk I/O and decode operations, you keep only a single bitmap in memory instead of multiple copies. This reduces CPU usage, which helps preserve battery life, and lowers the risk of crashes from memory pressure.
```

Comment:
We can also mention that with shared objects you're able to create much more convenient API that is more object-oriented (different than a stateless and a "flat" module object with only some functions, without classes).